agent/agenttest/agent.go

```diff
@@ -0,0 +1,147 @@
+package agenttest
+
+import (
+	"context"
+	"net/url"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/slogtest"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/coder/coder/v2/agent"
```

Comment:
Imports seem a bit wonkily organized (same in other files), likely due to how VSCode auto adds an import.

```suggestion
	"github.com/google/uuid"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"

	"cdr.dev/slog"
	"cdr.dev/slog/sloggers/slogtest"
	"github.com/coder/coder/v2/agent"
```